tests/test_inference_problem.py

```diff
@@ -0,0 +1,253 @@
+        if self.N == 1:
+            idx_range = str(self.start_idx)
+        else:
+            idx_range = f"{self.start_idx}..{self.start_idx+N}"
+
+        return f"{idx_range:10} {self._affected_parameters} {self.prior}"
+
+
+class LatentParameterList:
+    def __init__(self):
+        self._all_parameter_lists = {}
+        self._latent = OrderedDict()
+        self._total_length = None
+
+    def add_parameter_list(self, parameter_list, key=None):
+        assert key not in self._all_parameter_lists
+        self._all_parameter_lists[key] = parameter_list
+
+    def add(self, latent_name, parameter_name, key=None):
+        assert key in self._all_parameter_lists
+        assert self._all_parameter_lists[key].has(parameter_name)
+
+        try:
+            N = len(self._all_parameter_lists[key][parameter_name])
+        except:
+            N = 1
+        
+        if latent_name not in self._latent:
+            self._latent[latent_name] = LatentParameter()
+
+        self._latent[latent_name].add(key, parameter_name, N)
+        self._update_idx()
+
+    def add_by_name(self, latent_name):
+        for key, prm_list in self._all_parameter_lists.items():
+            if prm_list.has(latent_name):
+                self.add(latent_name, latent_name, key)
+        
+
+    def _update_idx(self):
+        self._total_length = 0
+        for key, latent in self._latent.items():
+            latent.start_idx = self._total_length
+            self._total_length += latent.N or 0
+
+    def update(self, number_vector):
+        assert len(number_vector) == self._total_length
+        for l in self._latent.values():
+            latent_numbers = number_vector[l.start_idx : l.start_idx + l.N]
+            for (key, prm_name) in l:
+                self._all_parameter_lists[key][prm_name] = latent_numbers
+
+    def __str__(self):
+        s = ""
+        for key, latent in self._latent.items():
+            s += f"{key:10}: {latent} \n"
+        return s
+
+
+class Noise:
+    def __init__(self):
+        self._noise_groups = OrderedDict()
+        self._noise_prior = OrderedDict()
+
+    def add(self, noise_name, sensor, key=None):
+        if noise_name not in self._noise_groups:
+            self._noise_groups[noise_name] = []
+
+        noise_group = self._noise_groups[noise_name]
+        assert (key, sensor) not in noise_group
+
+        for (_, existing_sensor) in noise_group:
+            assert existing_sensor.shape == sensor.shape
+
+        noise_group.append((key, sensor))
+
+
+"""
+###############################################################################
+
+                            USER CODE
+
+###############################################################################
+"""
+
+class MySensor(Sensor):
+    def __init__(self, name, position):
+        super().__init__(name, shape=(1,1))
+        self.position = position
+
+class MyForwardModel:
+    def __call__(self,parameter_list, sensors, time_steps):
```

Comment:
In our demotwin, it made sense to separate the time steps. Conceptually, the sensor is really "just there" (i.e. attached somewhere) and can measure stuff. The measurement system then associates the sensor values with time information. Translated to our library, the model itself depends on the time information (e.g. time dependent boundary conditions) and the sensor, again, just measures. 

Practically, I can use the same sensor in multiple measurements where the time steps (I actually meant `time_stamps` or `times`, but that does not matter much) differ.

Reply:
That is true, but image the case of having two different sensors with different measurement frequencies. What are you returning in the output for each individual sensor, since this should not be the union of all sensor time steps. If so, you would have to manually decompose the results, since each sensor type has only data for a subset of all time steps.

Reply:
Maybe we can skip part of the discussion here, as a forward model interface is not part of the library. We start at the "model_error". 

sensor1: 1Hz
sensor2: 100Hz

If I chose to evaluate that within the same forward_model or model_error, both would return (measured over 1 seconds) a dict like:
`{sensor1: <vector length 1>, sensor2: <vector length 100>}`

I do not understand where a union or decomposition is needed.